rust/windows-client/src-tauri/src/main.rs

```diff
@@ -1,237 +1,66 @@
-                    }
-                }
-            })
-            .build(tauri::generate_context!())
-            .expect("error while building tauri application")
-            .run(|_app_handle, event| {
-                if let tauri::RunEvent::ExitRequested { api, .. } = event {
-                    // Don't exit if we close our main window
-                    // https://tauri.app/v1/guides/features/system-tray/#preventing-the-app-from-closing
-
-                    api.prevent_exit();
-                }
-            });
-    }
-
-    pub fn main_debug_wintun() {
-        use std::sync::Arc;
-
-        //Must be run as Administrator because we create network adapters
-        //Load the wintun dll file so that we can call the underlying C functions
-        //Unsafe because we are loading an arbitrary dll file
-        let wintun = unsafe { wintun::load_from_path("../wintun/bin/amd64/wintun.dll") }
-            .expect("Failed to load wintun dll");
-
-        //Try to open an adapter with the name "Demo"
-        let adapter = match wintun::Adapter::open(&wintun, "Demo") {
-            Ok(a) => a,
-            Err(_) => {
-                //If loading failed (most likely it didn't exist), create a new one
-                wintun::Adapter::create(&wintun, "Demo", "Example manor hatch stash", None)
-                    .expect("Failed to create wintun adapter!")
-            }
-        };
-        //Specify the size of the ring buffer the wintun driver should use.
-        let session = Arc::new(adapter.start_session(wintun::MAX_RING_CAPACITY).unwrap());
-
-        //Get a 20 byte packet from the ring buffer
-        let mut packet = session.allocate_send_packet(20).unwrap();
-        let bytes: &mut [u8] = packet.bytes_mut();
-        //Write IPV4 version and header length
-        bytes[0] = 0x40;
-
-        //Finish writing IP header
-        bytes[9] = 0x69;
-        bytes[10] = 0x04;
-        bytes[11] = 0x20;
-        //...
-
-        //Send the packet to wintun virtual adapter for processing by the system
-        session.send_packet(packet);
-
-        println!("Sleeping 1 minute, see if the adapter is visible...");
-        std::thread::sleep(std::time::Duration::from_secs(60));
-
-        //Stop any readers blocking for data on other threads
-        //Only needed when a blocking reader is preventing shutdown Ie. it holds an Arc to the
-        //session, blocking it from being dropped
-        session.shutdown().unwrap();
-
-        //the session is stopped on drop
-        //drop(session);
-
-        //drop(adapter)
-        //And the adapter closes its resources when dropped
-    }
-
-    // Learn more about Tauri commands at https://tauri.app/v1/guides/features/command
-    #[tauri::command]
-    fn greet(name: &str) -> String {
-        format!("Hello, {}! You've been greeted from Rust!", name)
+    // Special case for app link URIs
+    if let Some(arg) = std::env::args().nth(1) {
+        if arg.starts_with("firezone://") {
```

Comment:
```suggestion
        if arg.starts_with("firezone-fd0020211111://") {
```

Reply:
Per standup, I'll make this change and leave a TODO noting that we can't ship until the portal encrypts the response because the app links could be captured by malware.

I'll also extract this to a const str, I think it's duplicated in two places

Reply:
Done in 9c07629b